capstone/src/main/webapp/authenticate.js

```diff
@@ -59,11 +51,12 @@ function initClient() {
         discoveryDocs: DISCOVERY_DOCS,
         scope: SCOPES
     }).then(function () {
+        // Set ID_TOKEN and dispatch event
+        window.ID_TOKEN = gapi.auth2.getAuthInstance().currentUser.get().getAuthResponse().id_token;
+        window.dispatchEvent(new Event('authorized'));
```

Comment:
Another (preferable) alternative is to add the token to the event itself. Relying on global variables is sometimes worth the convenience, but in this case you can set the id token on the authorized event itself, and then anything which needs the token MUST listen to the authorized event, and cant mistakenly just assume that somehow the global will be set in time before xyz JS is run